.builder/actions/pkcs11_test_setup.py

```diff
@@ -11,19 +11,38 @@
 class Pkcs11TestSetup(Builder.Action):
     """
     Set up this machine for running the PKCS#11 tests.
-    If SoftHSM2 is not installed, the tests are skipped.
+    If SoftHSM2 cannot be installed, the tests are skipped.
 
-    This action should be run in the 'test_steps' stage.
+    This action should be run in the 'pre_build_steps' or 'build_steps' stage.
     """
 
     def run(self, env):
         self.env = env
 
+        # total hack: don't run PKCS#11 tests when building all C libs with -DBUILD_SHARED_LIBS=ON.
+        # here's what happens:  libsofthsm2.so loads the system libcrypto.so and
+        # s2n loads the aws-lc's libcrypto.so and really strange things start happening.
+        # this wouldn't happen in the real world, just in our tests, so just bail out
+        if any('BUILD_SHARED_LIBS=ON' in arg for arg in env.args.args):
+            print("WARNING: PKCS#11 tests disabled when BUILD_SHARED_LIBS=ON due to weird libcrypto.so behavior")
+            return
+
+        # try to install softhsm
+        try:
+            softhsm_install_acion = Builder.InstallPackages(['softhsm'])
+            softhsm_install_acion.run(env)
+        except:
+            print("WARNING: softhsm could not be installed. PKCS#11 tests are disabled")
```

Comment:
should this read "SoftHSM2" instead of "softhsm"?

Reply:
Was just a cut and paste of Mike's aws-c-io script which supposedly is more forgiving.